polyfill/lib/calendar.mjs

```diff
@@ -672,6 +675,14 @@ const nonIsoHelperBase = {
         result.era = value;
       }
     }
+    if (hasEra && !result.era) {
+      // ICU bug that neglects to provide an era code for negative eraYear in
+      // the coptic calendar
+      assert(this.id === 'coptic', 'Unable to discern era from Intl.DateTimeFormat.formatToParts in Coptic calendar.');
```

Comment:
In other places where we bump into ICU bugs or limitations, we throw a RangeError. Should we do the same here?